python/housinginsights/sources/cama.py

```diff
@@ -0,0 +1,208 @@
+        don't provide any housing units and bedrm at this time.
+        """
+        for row in cama_data['features']:
+            try:
+                current_year = int(datetime.date.today().strftime('%Y'))
+                #Skipping none values for units under construction
+                if row['properties']['AYB'] is not None and int(row['properties']['AYB']) > current_year:
+                    continue
+
+                objectid = row['properties']['OBJECTID']
+                if len(row['properties']['SSL']) == 8:
+                    square = row['properties']['SSL'][:4]
+                    lot = row['properties']['SSL'][4:]
+                else:
+                    square, lot = row['properties']['SSL'].split()
+                suffix = ' '
+                if len(square) > 4:
+                    square = square[:4]
+                    suffix = square[-1]
+
+                mar_return = mar_api.get_data(square, lot, suffix)
+
+                ''' Count the housing units and bedrooms '''
+                num_units = 0
+                if row['properties']['NUM_UNITS']: num_units = row['properties']['NUM_UNITS']
+                if num_units == 0:
+                    num_units = 1
+
+                bedrm = row['properties']['BEDRM']
+                if bedrm == 0: bedrm = 1
+                if bedrm == None: bedrm = 0
+
+                for zone in zone_types:
+                    if zone == 'anc': zone_count = anc_count
+                    elif zone == 'census_tract': zone_count = census_count
+                    elif zone == 'neighborhood_cluster': zone_count = cluster_count
+                    elif zone == 'ward': zone_count = ward_count
+                    elif zone == 'zip': zone_count = zipcode_count
+
+                    if 'Warning' not in mar_return.keys():
+                        flag = False
+                        for dictionary in zone_count: #dictionary is {'zone_type': 'ANC', 'zone': 'ANC 8A', etc.}
+                            if dictionary['zone'] == mar_return[zone]: #mar_return[ANC] is 'ANC 8A'
+                                dictionary['housing_unit_count'] += num_units
+                                dictionary['bedroom_unit_count'] += bedrm
+                                flag = True
+                                break
+                        if not flag:
+                            zone_count.append( OrderedDict([('zone_type', zone), ('zone', mar_return[zone]), ('housing_unit_count', num_units), ('bedroom_unit_count', bedrm)]) )
+
+            except Exception as e:
+                exc_type, exc_obj, exc_tb = sys.exc_info()
+                print(exc_type, "line", exc_tb.tb_lineno)
+                print("Error! SSL: ", row['properties']['SSL'], row['properties']['AYB'])
+                continue
+
+        return {'anc': anc_count, 'census_tract': census_count, 'neighborhood_cluster': cluster_count, 'ward': ward_count, 'zip': zipcode_count}
```

Comment:
During the refactor, to get this to behave like other ApiConn objects, the 'get_data' should be the one that produces the csv.